File: packages/frontend/src/api/auth.api.ts
import api from './client';
import type { ApiResponse, User } from '@/types';

export const authApi = {
  sendOtp: (phone: string) =>
    api.post<ApiResponse<{ message: string }>>('/auth/send-otp', { phone }).then((r) => r.data.data),

  verifyOtp: (phone: string, code: string, role?: 'CUSTOMER' | 'VENDOR' | 'TRADER') =>
    api.post<ApiResponse<{ token: string; user: User; isNewUser: boolean }>>('/auth/verify-otp', { phone, code, role }).then((r) => r.data.data),

  getProfile: () =>
    api.get<ApiResponse<User>>('/auth/me').then((r) => r.data.data),

  updateProfile: (data: {
    name?: string;
    businessName?: string;
    address?: string;
    role?: 'CUSTOMER' | 'VENDOR' | 'TRADER';
  }) => api.put<ApiResponse<User>>('/auth/me', data).then((r) => r.data.data),
};

File: packages/mobile/lib/core/api/auth_api.dart
import 'package:dio/dio.dart';

import '../network/api_response.dart' show ApiResponse, ApiEnvelopeException;
import '../../shared/models/user.dart';

class AuthApi {
  AuthApi(this._dio);

  final Dio _dio;

  Future<void> sendOtp(String phone) async {
    final res = await _dio.post<Map<String, dynamic>>('/auth/send-otp', data: {'phone': phone});
    final body = res.data!;
    final r = ApiResponse.parseMap(body);
    if (!r.success) {
      throw ApiEnvelopeException(r.message.isNotEmpty ? r.message : 'Request failed');
    }
  }

  Future<({String token, User user, bool isNewUser})> verifyOtp({
    required String phone,
    required String code,
    String? role,
  }) async {
    final res = await _dio.post<Map<String, dynamic>>(
      '/auth/verify-otp',
      data: {
        'phone': phone,
        'code': code,
        'role': ?role,
      },
    );
    return ApiResponse.unwrapData(res.data!, (data) {
      final m = data as Map<String, dynamic>;
      return (
        token: m['token'] as String,
        user: User.fromJson(m['user'] as Map<String, dynamic>),
        isNewUser: m['isNewUser'] == true,
      );
    });
  }

  Future<User> getProfile() async {
    final res = await _dio.get<Map<String, dynamic>>('/auth/me');
    return ApiResponse.unwrapData(res.data!, (data) => User.fromJson(data as Map<String, dynamic>));
  }

  Future<User> updateProfile(Map<String, dynamic> body) async {
    final res = await _dio.put<Map<String, dynamic>>('/auth/me', data: body);
    return ApiResponse.unwrapData(res.data!, (data) => User.fromJson(data as Map<String, dynamic>));
  }
}
